float64[] pos
float64 vel
---
bool ret
